feat(prometheus/operator): accept relabel rules exported from `prometheus.relabel` (#6830)
grafana/agent: docs/sources/flow/reference/components/prometheus.operator.podmonitors.md

---- | ---- | ----------- | ------- | --------
`forward_to` | `list(MetricsReceiver)` | List of receivers to send scraped metrics to. | | yes
`namespaces` | `list(string)` | List of namespaces to search for PodMonitor resources. If not specified, all namespaces will be searched. || no
`relabel_rules` | `RelabelRules` | Relabeling rules to apply on scraped metric data points. | `{}` | no

> The `relabel_rules` argument can make use of the `rules` export value from a
> [prometheus.relabel][] component to apply one or more relabeling rules to metric data points
> before they're forwarded to the list of receivers in `forward_to`.

Reviewer: ```suggestion
The `relabel_rules` argument uses the `rules` export value from a [prometheus.relabel][] component to apply one or more relabeling rules to your metric data points. The metric data points are relabeled and forwarded to the list of receivers in `forward_to`.
```

A small rework to make the information active instead of passive. Is it still technically accurate with these changes?

If this is correct, this change shoudl be applied to teh other cases here in each of the following doc topics.
